test/boost/conftest.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import sys
+from pathlib import Path, PosixPath
+
+import pytest
+from pytest import Collector
+
+from test.pylib.cpp.boost.boost_facade import BoostTestFacade
+from test.pylib.cpp.boost.prepare_instance import get_prepare_instance
+from test.pylib.cpp.common_cpp_conftest import collect_items, get_modes_to_run
+
+pytest_plugins = [
+    "test.pylib.cpp.common_cpp_conftest",
+]
+
+
+def pytest_collect_file(file_path: PosixPath, parent: Collector):
+    """
+    Method triggered automatically by pytest to collect files from a directory. Boost and unit have the same logic for
+    collection, the only difference in execution, and it's covered by facade
+    """
+    return collect_items(file_path, parent, facade=BoostTestFacade(parent.config),run_id=parent.config.getoption('run_id'))
+
+
+@pytest.hookimpl(wrapper=True)
+def pytest_runtestloop(session):
+    """
+    https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_runtestloop
+    This hook is needed to start the Minio and S3 mock servers before tests. After starting the servers, the default
+    pytest's runtestloop takes control. Finally part is responsible for stopping servers regardless of failure in the tests.
+    """
+    modes = get_modes_to_run(session)
+    is_worker = False
+    if 'xdist' in sys.modules:
+        sys.modules['xdist'] .is_xdist_worker(session)
+    temp_dir = Path(session.config.getoption('tmpdir'))
+
+    prepare_instance = get_prepare_instance(temp_dir, is_worker, modes)
+    prepare_instance.prepare()
```

Comment:
Rewrote the docstring, is better now?

Reply:
I think prepare_instance is a bit confusing name, because "prepare" is a verb but is used as an adjective.

Reply:
I called it instance because we have two variants:
1. with xdist we need to prepare the main process and child processes differently.
2. without xdist we need to prepare only the main process, so I can't refer as xdist process because there is no xdist. but it has to prepare the same thing as the main xdist process.